src/libpsl-native/src/getppid.cpp

```diff
@@ -0,0 +1,39 @@
+#include "getppid.h"
+
+#include <sys/user.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+
+//! @brief GetPPid returns the parent process id for a process
+//!
+//! GetPPid
+//!
+//! @param[in] pid
+//! @parblock
+//! The process id to query for it's parent.
+//! @endparblock
+//!
+//! @retval the parent process id, or UINT_MAX if unsuccessful
+//!
+pid_t GetPPid(pid_t pid)
+{
+
+#if defined(__APPLE__) && defined(__MACH__)
+
+    const pid_t PIDUnknown = UINT_MAX;
+    struct kinfo_proc info;
+    size_t length = sizeof(struct kinfo_proc);
+    int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, pid};
+    if (sysctl(mib, 4, &info, &length, NULL, 0) < 0)
+        return PIDUnknown;
+    if (length == 0)
+        return PIDUnknown;
+
+    return info.kp_eproc.e_ppid;
+
+#else
+
+    return UINT_MAX;
```

Comment:
This would be a developer problem: if a Linux dev accidentally tries to use `GetPPid`, as is, they function will be called correctly, but only ever return `UINT_MAX`, which to me seems like it may be more confusing than an exception stating it couldn't call the function at all. Either way I think they'll end up looking at the native source pretty quickly, and should be able to figure out their problem (to use the `/proc` version instead). But I don't think a user is going to run into this.